src/models/rule_graph.rs

```diff
@@ -54,6 +55,15 @@ pub struct RuleGraph {
   graph: HashMap<String, Vec<(String, String)>>,
 }
 
+impl Validator for RuleGraph {
+  fn validate(&self) -> Result<(), String> {
+    match self.rules().iter().try_for_each(|rule| rule.validate()) {
+      Ok(()) => Ok(()),
+      Err(e) => Err(format!("Incorrect Rule Graph - {}", e)),
```

Comment:
Question: Do we report the first failing rule or all syntax errors in every rule in the rule graph? (Neither would be right or wrong, specially at this point, just want to make sure. I'd think it's the first case looking at this, but nor sure what the semantics of `try_for_each` are.

Reply:
We are short-circuiting at the first exception. 
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.try_for_each